### PR TITLE
fix: support some virtual contexts in `toThrow`

### PR DESCRIPTION
**Pull request by 43081j**
This adds support for VM situations where we pass a `RegExp` from another process.

Note that we don't have a full fix for this stuff until `check-error` also supports `Error` being from another origin.

**There's also a tiny fix in here for supporting `throw ''`.** I have left a PR comment by the relevant line of code

### CODE REVIEW

File: web-test-runner.config.js [hunk 1]
```diff
@@ -5,7 +5,10 @@ const commonjs = fromRollup(rollupCommonjs);
 
 export default {
   nodeResolve: true,
-  files: ["test/*.js"],
+  files: [
+    "test/*.js",
+    "!test/virtual-machines.js"
```
[the remark this answers is not in the document]
Comment (43081j) [reply web-test-runner.config.js:10]: good catch! i forgot to `git add` it

**Review by keithamus — APPROVED** [review]
LGTM

File: lib/chai/utils/index.js [hunk 2]
```diff
@@ -94,3 +94,7 @@ export {isNaN} from './isNaN.js';
 
 // getOperator method
 export {getOperator} from './getOperator.js';
+
+export function isRegExp(obj) {
+  return Object.prototype.toString.call(obj) === '[object RegExp]';
+}
```
Comment (koddsson) [inline lib/chai/utils/index.js:100]: Why is this needed?

**Review by koddsson — APPROVED** [review]
Looks good :)

File: lib/chai/utils/index.js (hunk 2 above)
Comment (43081j) [reply lib/chai/utils/index.js:100]: A RegExp from a virtual context isn't the same RegExp as in the owning context, so instanceof won't work

If you pass chai into the virtual context, it'll be comparing against it's own RegExp rather than that of the virtualised code

**Comment by 43081j** [comment]
@koddsson could you re-review this? i updated slightly to support throwing of _anything_ (so even `throw undefined` will work)

and i wrapped `getConstructorName` in a try/catch since it may be possible we pass a non-constructable function to it

File: test/virtual-machines.js [hunk 3]
```diff
@@ -0,0 +1,31 @@
...
+}
+
+describe('node virtual machines', function () {
+  it('throws', function() {
+    const shouldNotThrow = [
+      `assert.throws(function() { throw ''; }, /^$/);`,
+      `assert.throws(function() { throw new Error('bleepbloop'); });`,
+      `assert.throws(function() { throw new Error(''); });`,
+      // TODO (43081j): enable this test once check-error supports
+      // cross-vm `Error` objects
```
Comment (koddsson) [inline test/virtual-machines.js:19]: Could we link to a issue on `chaijs/check-error` so it's easy to follow this work. In case this doesn't get done in the near future we then have captured the relevant context. If you got a PR practically ready then no worries of course.

**Review by koddsson — APPROVED** [review]
Looks good to me! I have one question but happy to merge whatever the answer to that is.

File: test/virtual-machines.js (hunk 3 above)
Comment (43081j) [reply test/virtual-machines.js:19]: this is the one you published today 🎉 

i just need to update the branch i think and remove this comment

Comment (koddsson) [reply test/virtual-machines.js:19]: Oh I see haha 👯

**Review by koddsson — APPROVED** [review]
Looks good!
